src/state_history_kv_tables.hpp

```diff
@@ -41,10 +41,11 @@ inline bool operator==(const fill_status_v0& a, fill_status_v0& b) {
 
 inline bool operator!=(const fill_status_v0& a, fill_status_v0& b) { return !(a == b); }
 
-struct fill_status_kv : eosio::table<fill_status> {
-   index primary_index{ eosio::name{ "primary" }, [](const auto& var) { return std::vector<char>{}; } };
+struct fill_status_kv : eosio::kv_table<fill_status> {
+   index<std::vector<char>> primary_index{ eosio::name{ "primary" },
```

Comment:
Is there a singleton table type available?

Reply:
Yes, I could clean the singleton work up a bit and bring it in here.